code/modules/mob/inventory.dm

```diff
@@ -220,7 +220,8 @@
 	if(held_items[index])
 		return 0
 
-	if(!Adjacent(W) && isturf(W.loc) && !W.arcanetampered)
+	// adjacent only loc checks one way around, so first part is necessary
+	if(W.loc != src && !Adjacent(W) && isatom(W.loc) && !W.arcanetampered)
```

Comment:
Is the ``isatom(W.loc)`` check specifically for making sure that the loc isn't null? There's been some discussion on why it's here.

Reply:
yea, i coded it so nullspace spawns ignore all this logic at the time (like in datum outfit stuff)

Reply:
Wouldn't an !isnull() be faster? Or, even better, a flat W.loc?

Reply:
i think i made it like that because it was inside some datums too

Reply:
Reference and quick testing confirms that loc can only ever be an atom or null. Attempting to force it to be anything else will set it to null (or worse, runtime).

Reply:
interesting, removing this check since the recursive loc one i added makes it redundant